bika/lims/browser/resultsimport/resultsimport.py

```diff
@@ -0,0 +1,249 @@
+                             if isfile(join(folder, f))]
+                imported_list = self.getAlreadyImportedFiles(folder)
+                if not imported_list:
+                    logger.warn('imported.csv file not found ' + interface)
+                    self.add_to_logs(i, interface,
+                                     'imported.csv File not found...', '')
+                    continue
+                for file_name in all_files:
+                    if file_name in imported_list:
+                        continue
+                    temp_file = open(folder+'/'+file_name)
+                    # Parsers work with UploadFile object from
+                    # zope.HTTPRequest which has filename attribute.
+                    # To add this attribute we convert the file.
+                    # CHECK should we add headers too?
+                    result_file = ConvertToUploadFile(temp_file)
+                    exim = instruments.getExim(interface)
+                    parser_name = instruments.getParserName(interface)
+                    parser_function = getattr(exim, parser_name) \
+                        if hasattr(exim, parser_name) else ''
+                    if not parser_function:
+                        self.add_to_logs(i, interface,
+                                         'Parser not found...', file_name)
+                        continue
+                    # We will run import with some default parameters
+                    # Expected to be modified in the future.
+                    logger.info('Parsing ' + file_name)
+                    parser = parser_function(result_file)
+                    importer = GeneralImporter(
+                                parser=parser,
+                                context=self.portal,
+                                idsearchcriteria=['getRequestID',
+                                                  'getSampleID',
+                                                  'getClientSampleID'],
+                                allowed_ar_states=['sample_received'],
+                                allowed_analysis_states=None,
+                                override=[False, False],
+                                instrument_uid=i.UID())
+                    tbex = ''
+                    try:
+                        importer.process()
+                    except:
+                        tbex = traceback.format_exc()
+                    errors = importer.errors
+                    logs = importer.logs
+                    if tbex:
+                        errors.append(tbex)
+                    final_log = ''
+                    success_log = self.getInfoFromLog(logs, 'Import finished')
+                    if success_log:
+                        final_log = success_log
+                    else:
+                        final_log = errors
+                    self.insert_file_name(folder, file_name)
+                    self.add_to_logs(i, interface, final_log, file_name)
```

Comment:
Only save the last 100 chars from `final_log` here in order to keep the database smaller